src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -800,6 +800,12 @@ def list_features(client, resource_provider_namespace=None):
     return client.list_all()
 
 
+def register_feature(client, resource_provider_namespace, feature_name):
+    logger.warning("Once the fearure '{}' is registered, invoking 'az provider register -n {}' is required "
+                   "to get the change propagated".format(resource_provider_namespace, feature_name))
+    return client.register(resource_provider_namespace, feature_name)
```

Comment:
I am in the conversation with ARM team for whether it is a service bug. If yes, I will revert the change